src/OSPSuite.UI/Views/SimulationOutputMappingView.cs

```diff
@@ -119,5 +131,37 @@ private RepositoryItem allOutputsRepository(SimulationOutputMappingDTO dto)
       public override string Caption => Captions.SimulationUI.ObservedDataSelection;
 
       public override ApplicationIcon ApplicationIcon => ApplicationIcons.ObservedData;
+
+      private void btnDeleteItemClick()
+      {
+         var selectedDTOs = gridView.GetSelectedRows().SelectMany(x => _gridViewBinder.SelectedItems(x)).ToList();
+
+         if (!selectedDTOs.Any())
+            return;
+
+         _presenter.RemoveObservedData(selectedDTOs);
+      }
+
+      private void onPopupMenuShowing(object sender, PopupMenuShowingEventArgs e)
+      {
+         if (e.Menu == null || e.HitInfo.HitTest != GridHitTest.RowIndicator)
+            return;
+
+         var selectedDTOs = gridView.GetSelectedRows().SelectMany(x => _gridViewBinder.SelectedItems(x)).ToList();
+
+         if (!selectedDTOs.Any())
+            return;
+
+         var deleteItem = new DevExpress.Utils.Menu.DXMenuItem(
+            Captions.DeleteSelected,
+            (o, args) => _presenter.RemoveObservedData(selectedDTOs)
+         )
+         {
+            Shortcut = System.Windows.Forms.Shortcut.Del,
+            SvgImage = ApplicationIcons.DeleteSelected
+         };
+
+         e.Menu.Items.Insert(2, deleteItem);
```

Comment:
If you intend to insert at the end, then can you first count the menu entries and then calculate the insert index? Someone might come along and change the number of items in the list